src/test/e2e/23_data_injection_test.go

```diff
@@ -31,15 +32,28 @@ func TestDataInjection(t *testing.T) {
 		runDataInjection(t, path)
 	}
 
+	c, err := cluster.NewCluster()
+	require.NoError(t, err)
+
+	// Find a running kiwix pod
+	pods, err := c.GetPods("kiwix", &metav1.ListOptions{
+		FieldSelector: "status.phase=Running",
+		LabelSelector: "app=kiwix-serve",
+	})
+	require.NoError(t, err)
+	var runningKiwixPod string
+	for _, pod := range pods.Items {
+		runningKiwixPod = pod.Name
+		break
+	}
+
 	// Verify the file and injection marker were created
-	stdOut, stdErr, err := e2e.Kubectl("--namespace=kiwix", "logs", "--tail=5", "--selector=app=kiwix-serve", "-c=kiwix-serve")
+	stdOut, stdErr, err := e2e.Kubectl("--namespace=kiwix", "logs", runningKiwixPod, "--tail=5", "-c=kiwix-serve")
```

Comment:
This seems like a lot for what could be `--field-selector=status.phase=Running`.  Going to just comment and let @Noxsios decide.